lib/v8-to-istanbul.js

```diff
@@ -112,22 +113,25 @@ module.exports = class V8ToIstanbul {
         if (this.excludePath(path)) {
           return
         }
-        const lines = covSource.lines.filter(line => {
-          // Upstream tooling can provide a block with the functionName
-          // (empty-report), this will result in a report that has all
-          // lines zeroed out.
-          if (block.functionName === '(empty-report)') {
+
+        // (empty-report), this will result in a report that has all
+        // lines zeroed out.
+        if (block.functionName === '(empty-report)') {
+          this.all = true
+          covSource.lines.forEach(line => {
             line.count = 0
-            this.all = true
-            return true
-          }
+          })
+        }
+
+        const lines = this.all ? covSource.lines : sliceRange(covSource.lines, startCol, endCol)
+        if (!lines.length) {
+          return
```

Comment:
I think this would need to be something closer to:

```js
        let lines
        if (block.functionName === '(empty-report)') {
          lines = covSource.lines.filter((line) => {
            line.count = 0
            this.all = true
            return true
          })
        } else {
          lines = sliceRange(covSource.lines, startCol, endCol)
        }
```

Reply:
thank you...that is definitely better than what I had originally... 
I just moved the assignment of this all out of the loop being careful to do
`this.all = lines.length > 0`
...sorry it's not premature optimisation it's OCD 😆